idpf_specification.md

```diff
@@ -6433,28 +6433,101 @@ VIRTCHNL2_OP_RDMA			= 529,
+VIRTCHNL2_PTP_CAP_TX_TSTAMPS	
+VIRTCHNL2_PTP_CAP_TX_TSTAMPS_MB
+The packet delay calculation on the wire is used to accurately remove the time component that is common in calculation of the skew in clock between the leader and the follower. Tx timestamping should be allowed on all vports that are tied to an external port which means there traffic goes only to a particular external physical port.
+3.	Ability to Adjust Device Master Clock, this capability in a multi-complex system is given to one of the CPU complex that is trusted to make this master clock adjustment.  The Complex that calculates the skew between the leader (master system in the PTP network domain)) and the current Device master clock (slave system in the PTP network domain) runs the PTP protocol and algorithm in SW which requires other capabilities such as TX_TSTAMPS and cross timestamping as well from the Device as described in 1 and 2 above. 
+VIRTCHNL2_PTP_CAP_ADJ_DEVICE_CLK
+	VIRTCHNL2_PTP_CAP_ADJ_DEVICE_CLK_MB
+In a system this can be done through direct register writes or over mailbox. If done over mailbox (_MB), the mailbox latency should be constant and adjusted for when setting the device time to be as close in time to the leader as possible. 
+4.	Ability to initialize the Device Master clock, this is done at the start of the server and is done by a trusted entity for the system. Again this can be done directly or over mailbox (_MB), when doing it over mailbox similar consideration for adjustment in mailbox latency should be made.
+VIRTCHNL2_PTP_CAP_SET_DEVICE_CLK_TIME		
+VIRTCHNL2_PTP_CAP_SET_DEVICE_CLK_TIME_MB
+5.	Ability to do Rx Tiemstamping of packets as they arrive on the device, The Device automatically latches timestamp information on the receive path, and the timestamp is carried through the data pipeline in the device as metadata. Using the default 32 byte descriptors as defined in the spec the driver always gets the rx_timestamp in the descriptor as a 40 bit value spread in 3 fields. The driver can check if the rx_timestamp field is valid or not by lowest bit in ts_ns field in the descriptor being 1 or not.  If the lowest bit is set, then the timestamp value is valid for the default 32 byte descriptor.
+There is no explicit enabling or disabling of this capability.
+Advanced RX descriptors when defined for the device for newer capabilities may override the upper 32 bit rx_timestamp fields with other fields and indicate the same with a new RXDID which will define the new field layout for the flex fields. This would require the driver to use the right descriptor layout and adjust the flow accordingly.
+
+
+### Configuration
+
+PTP Virtchannel Commands
+	VIRTCHNL2_OP_PTP_GET_CAPS		
+	VIRTCHNL2_OP_PTP_GET_VPORT_TX_TSTAMP	
+	VIRTCHNL2_OP_PTP_GET_DEV_CLK_TIME
+	VIRTCHNL2_OP_PTP_GET_CROSS_TIME	
+	VIRTCHNL2_OP_PTP_SET_DEV_CLK_TIME
+	VIRTCHNL2_OP_PTP_ADJ_DEV_CLK_FINE
```

Comment:
This command is not described by any of the capabilities above. It's different that device clock set/adj because it does not directly modify the clock value, just an increment rate.

Reply:
Its described as part of time frequency adjustment.